src/providers/NetworkProvider.tsx

```diff
@@ -19,7 +19,7 @@ export const NetworkProvider = ({ children }: Props) => {
 
     // Initial network check to consume initial "null" values
     // and avoid UI glitches due to initial network state
-    NetInfo.fetch();
+    NetInfo.fetch().then(setNetworkInfo);
```

Comment:
given the above comment on line 20 it seems this call was meant to be here just to consume the initial "null" values, are we sure we should be calling `setNetworkInfo` here?